account.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"math"
+	"time"
+
+	"github.com/cockroachdb/apd/v2"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// Account is an internal representation of a genesis regen account
+type Account struct {
+	Address       sdk.AccAddress
+	TotalRegen    apd.Decimal
+	StartTime     time.Time
+	Distributions []Distribution
+	EndTime       time.Time
```

Comment:
Could this just be a function that computes from the max of all distribution `dist.Time` values?

Separately from here, we'll need to deal w/ the logic of combining multiple account entries (from different SAFTs for the same user in the spreadsheet), like a reducer. II can imagine would be nice to not have `EndTime` as a struct field so we don't need to worry about setting it in the reduce function.

Reply:
Right, we may not need this as a separate field. I have code that does the merge functionality, will open as a separate PR soon